hudi-spark/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -49,6 +49,7 @@ object DataSourceReadOptions {
   val QUERY_TYPE_SNAPSHOT_OPT_VAL = "snapshot"
   val QUERY_TYPE_READ_OPTIMIZED_OPT_VAL = "read_optimized"
   val QUERY_TYPE_INCREMENTAL_OPT_VAL = "incremental"
+  val QUERY_TYPE_MOR_INCREMENTAL_OPT_VAL = "mor_incremental"
```

Comment:
we need to guard this with a flag. query types are fundamental to design. I prefer not to overload them

Reply:
This option was supposed to distinguish when the user wants to run an incremental query on the MOR table but on the parquet files only. This was not really necessary because users can just define the timestamp range to achieve the same goal. So we can just use the tableType. If COW, use the old way, if MOR, use the new relation. Don't need an extra option anymore imo.